03-pipefilter.md

```diff
@@ -16,277 +16,465 @@ Now that we know a few basic commands,
 we can finally look at the shell's most powerful feature:
 the ease with which it lets us combine existing programs in new ways.
 We'll start with a directory called `molecules`
-that contains six files describing some simple organic molecules.
-The `.pdb` extension indicates that these files are in Protein Data Bank format,
-a simple text format that specifies the type and position of each atom in the molecule.
+so lets go into that directory and take a look at that directory:
 
 ~~~ {.bash}
-$ ls molecules
+$ cd molecules
+~~~
+~~~ {.bash}
+$ ls *.pdb
 ~~~
 ~~~ {.output}
-cubane.pdb    ethane.pdb    methane.pdb
-octane.pdb    pentane.pdb   propane.pdb
+aldrin.pdb
+ammonia.pdb
+ascorbic-acid.pdb
+benzaldehyde.pdb
+camphene.pdb
+cholesterol.pdb
+cinnamaldehyde.pdb
+citronellal.pdb
+codeine.pdb
+cubane.pdb
+cyclobutane.pdb
+cyclohexanol.pdb
+cyclopropane.pdb
+ethane.pdb
+ethanol.pdb
+ethylcyclohexane.pdb
+glycol.pdb
+heme.pdb
+lactic-acid.pdb
+lactose.pdb
+lanoxin.pdb
+lsd.pdb
+maltose.pdb
+menthol.pdb
+methane.pdb
+methanol.pdb
+mint.pdb
+morphine.pdb
+mustard.pdb
+nerol.pdb
+norethindrone.pdb
+octane.pdb
+pentane.pdb
+piperine.pdb
+propane.pdb
+pyridoxal.pdb
+quinine.pdb
+strychnine.pdb
+styrene.pdb
+sucrose.pdb
+testosterone.pdb
+thiamine.pdb
+tnt.pdb
+tuberin.pdb
+tyrian-purple.pdb
+vanillin.pdb
+vinyl-chloride.pdb
+vitamin-a.pdb
 ~~~
 
-Let's go into that directory with `cd` and run the command `wc *.pdb`.
-`wc` is the "word count" command:
-it counts the number of lines, words, and characters in files.
-The `*` in `*.pdb` matches zero or more characters,
-so the shell turns `*.pdb` into a complete list of `.pdb` files:
+> ## `.pdb` Extension {.callout}
+>
+> The `.pdb` extension indicates that these files are in Protein Data Bank format,
+> a simple text format that specifies **at each line** the type and position of **one** atom in the molecule.
+
+The first question that we can ask is
+"how many files are inside the `molecules` directory?".
+If we could (1) write the output of `ls` into a file
+and (2) count the number of lines of that file
+we will have our answer.
+We are very luck and we can do (1) and (2) with the shell.
+For (1) we will use
 
 ~~~ {.bash}
-$ cd molecules
-$ wc *.pdb
-~~~
-~~~ {.output}
-  20  156 1158 cubane.pdb
-  12   84  622 ethane.pdb
-   9   57  422 methane.pdb
-  30  246 1828 octane.pdb
-  21  165 1226 pentane.pdb
-  15  111  825 propane.pdb
- 107  819 6081 total
+$ ls *.pdb > content-of-molecules
```

Comment:
this is detailed, but i'd prefer something shorter than `content-of-molecules`, maybe `molecule-list`?  less to type is better for teaching (and learning)